app/jmeter/bitbucket.jmx

```diff
@@ -478,6 +478,47 @@ String gitTmpWorkspace = props.get(&quot;GIT_TMP_WORKSPACE&quot;);
 vars.put(&quot;REPO_FOLDER&quot;, gitTmpWorkspace + &quot;/clone-push-&quot; + vars.get(&quot;REPO_SLUG&quot;) + &quot;-&quot; + UUID.randomUUID());</stringProp>
             </JSR223PreProcessor>
             <hashTree/>
+            <JSR223PreProcessor guiclass="TestBeanGUI" testclass="JSR223PreProcessor" testname="User git config global" enabled="true">
+              <stringProp name="scriptLanguage">groovy</stringProp>
+              <stringProp name="parameters"></stringProp>
+              <stringProp name="filename"></stringProp>
+              <stringProp name="cacheKey">true</stringProp>
+              <stringProp name="script">import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+Runtime rt = Runtime.getRuntime();
+        Process p = rt.exec(&quot;git config -l&quot;);
+
+        BufferedReader reader = new BufferedReader(
+                new InputStreamReader(p.getInputStream()));
+
+        String line;
+        boolean name = false;
+        boolean email = false;
+
+        while ((line = reader.readLine()) != null) {
+            if (line.contains(&quot;user.name&quot;)) {
+                if (line.endsWith(&quot;=&quot;)) {
```

Comment:
You just check if a user name is empty but It seems like it will not work if it is " " or "      ".
Also read all config is redundant.
You can execute `git config --global user.name` and `git config --global user.email` to get the value and check only them separately instead of iterating over the whole config. In this case please use `org.apache.commons.lang3.StringUtils.isBlank()` Jmeter has this lib in classpath. But for local testing you can download it from here https://mvnrepository.com/artifact/org.apache.commons/commons-lang3

Reply:
Thx for the advice!!!